db_sql.go

```diff
@@ -58,6 +62,12 @@ func (x *sqlUserStoreDB) Authenticate(identity, password string, authTypeCheck A
 		return ErrIdentityAuthNotFound
 	}
 
+	var accountLocked bool
+	row = x.db.QueryRow(`SELECT accountlocked FROM authuserpwd WHERE userid = $1`, userId)
+	if err := row.Scan(&accountLocked); err != nil || accountLocked {
```

Comment:
What are you doing about NULLs here? Rather use sql.NullBool and make the behaviour explicit and crystal clear.
Also, just write `x.db.QueryRow().Scan()`